server/apps/location.go

```diff
@@ -1,24 +1,86 @@
 package apps
 
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
 type LocationType string
 
 const (
-	LocationPostMenuItem = "post_menu_item"
+	LocationPostMenuItem      LocationType = "post_menu_item"
+	LocationChannelHeaderIcon LocationType = "channel_header_icon"
 )
 
-type Location interface {
-	Type() LocationType
+type LocationInt interface {
+	GetType() LocationType
 }
 
-type location struct {
+type Location struct {
 	LocationType LocationType
 	LocationID   string
+	Wish         Wish
```

Comment:
Or possibly on the `Wish` struct?

Reply:
We probably do, since we will performing the wish with the client for that app. I will think it further to see whether it should go in the Wish or the Location.